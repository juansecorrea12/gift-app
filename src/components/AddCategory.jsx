import { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({onNewCategory}) => {

    const[inputValue, setinputValue] = useState('');

    const handleInputChange = (event) => {
        setinputValue(event.target.value);
    }
    const onSubmit = (event) => {
        event.preventDefault();
        const newInputValue = inputValue.trim();
        if(newInputValue.length <= 1) return;
        onNewCategory(newInputValue)
        setinputValue('');
    }
    return(
        <>
        <form onSubmit={onSubmit} className="form-container" aria-label="form">
            <label 
                htmlFor="categoryName">
                Añade una categoria
            </label>
            <input 
                placeholder="Buscar Gift"
                type="text"
                value={inputValue}
                onChange = {handleInputChange} />
        </form>
        </>
    )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired,
}