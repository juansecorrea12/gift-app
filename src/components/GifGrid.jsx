import { GifGridItem } from "./GifGridItem";
import { useFetchGif } from "../hooks/useFetchGif";
import PropTypes from 'prop-types';

export const GifGrid = ({ category }) => {
    
    const {images, isLoading} = useFetchGif( category );

    return(
        <>
            <div className="category-element">
                <h1 className="category-name">{category}</h1>
                    {
                        isLoading && (<h2>Cargando...</h2>)
                    }
                <div className="card-grid">
                    {
                        images.map((image) => (
                            <GifGridItem 
                            key={image.id} 
                            { ...image } />
                        ))
                    }
                </div>
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
}
