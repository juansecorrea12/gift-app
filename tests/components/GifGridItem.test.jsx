import {render, screen} from '@testing-library/react';
import {GifGridItem} from '../../src/components/GifGridItem';

describe('Pruebas en el componente <GifGridItem />', () => {
    
    const title = 'vegeta';
    const url = 'https://vegeta-sayayin-gif.com/sayanVegeta.jpg';

    test('Debe de hacer match con el snapshot', () => {
        
        const {container} = render( <GifGridItem title = {title} url={url}/>);
        expect(container).toMatchSnapshot();
    })

    test('Debe mostrar la imagen con el URL indicado y el ALT indicado', () => {
        
        render( <GifGridItem title = {title} url={url}/>);
        // screen.debug();
        const {src, alt} = screen.getByRole('img');

        expect(src).toBe(url);
        expect(alt).toBe(title);
        expect(typeof src).toEqual("string");

    })

    test('Debe de mostrar el titulo en el componente <GifGridItem />', () => {
        render( <GifGridItem title = {title} url={url}/>);

        expect(screen.getByText( title )).toBeTruthy();


    })

})
