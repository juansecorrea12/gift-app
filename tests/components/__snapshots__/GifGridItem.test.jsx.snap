// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <GifGridItem /> Debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="vegeta"
      src="https://vegeta-sayayin-gif.com/sayanVegeta.jpg"
    />
    <p>
      vegeta
    </p>
  </div>
</div>
`;
